### ISSUE TITLE
Add upcoming GPT-3 model

### ISSUE THREAD

**Issue by stefan-it** [huggingface/transformers#4658]
# 🌟 New model addition

## Model description

The GPT-3 paper just landed on ArXiv: https://arxiv.org/abs/2005.14165.

Would be great to integrate it into Transformers, whenever models are available.

> Here we show that scaling up language models greatly improves task-agnostic, few-shot performance, sometimes even reaching competitiveness with prior state-of-the-art fine-tuning approaches. Specifically, we train GPT-3, an autoregressive language model with 175 billion parameters, 10x more than any previous non-sparse language model, and test its performance in the few-shot setting. For all tasks, GPT-3 is applied without any gradient updates or fine-tuning, with tasks and few-shot demonstrations specified purely via text interaction with the model. GPT-3 achieves strong performance on many NLP datasets, including translation, question-answering, and cloze tasks, as well as several tasks that require on-the-fly reasoning or domain adaptation, such as unscrambling words, using a novel word in a sentence, or performing 3-digit arithmetic. At the same time, we also identify some datasets where GPT-3's few-shot learning still struggles, as well as some datasets where GPT-3 faces methodological issues related to training on large web corpora. Finally, we find that GPT-3 can generate samples of news articles which human evaluators have difficulty distinguishing from articles written by humans. We discuss broader societal impacts of this finding and of GPT-3 in general.

## Open source status

* [x] GitHub repository is available: [here](https://github.com/openai/gpt-3)
* [ ] the model implementation is available: (give details)
* [ ] the model weights are available: (give details)
* [ ] who are the authors: (mention them, if possible by @gh-username)

**Comment by LouisCastricato** [comment]
My god, the paper hasn't even been up for a day...

**Comment by moinnadeem** [comment]
So who can run 175B parameters and what do I have to do for a favor?

**Comment by spronkoid** [comment]
whose pp do i have to diddle to run this model i can't imagine any scenario where i could run this myself lol

**Comment by AdamDanielKing** [comment]
The full model will be at least 350 GB (16-bit parameters). You'd need to partition it across more than (350 GB) / (16 GB) ~ **22 GPUs** just to run it! Not to mention the egress costs of making a model that size available.

Of course, the paper shows **8** different-sized models, **4 of which are smaller than GPT-2**, so some of those could be practical. 🙂 

![image](https://user-images.githubusercontent.com/5590173/83222642-ca2dc000-a146-11ea-934d-58001e955941.png)

**Comment by GraphGrailAi** [comment]
Is there any Colab to test at least GPT-3 XL ?

**Comment by AdamDanielKing** [comment]
> Is there any Colab to test at least GPT-3 XL ?

They haven't released any code or pretrained models yet. See the issue on the official repo: https://github.com/openai/gpt-3/issues/1

**Comment by minimaxir** [comment]
Note that the released models may be FP16, which may require forcing FP16 for use/finetuning (and therefore hardware-limited), or casting up to FP32.

**Comment by andifunke** [comment]
> Of course, the paper shows **8** different-sized models, **4 of which are smaller than GPT-2**, so some of those could be practical. slightly_smiling_face

One of the main benefits of the smaller gpt-3 models compared to their gpt-2 counterparts could be the increased context length of 2048 tokens.

**Comment by enzoampil** [comment]
Yeah, personally, I wouldn't be able use the models that won't fit in a Tesla P100

**Comment by AdamDanielKing** [comment]
The [GPT-3 repo](https://github.com/openai/gpt-3) is now archived (read-only) so perhaps OpenAI isn't planning on releasing anything this time around.

**Comment by ljlueloff** [comment]
> The [GPT-3 repo](https://github.com/openai/gpt-3) is now archived (read-only) so perhaps OpenAI isn't planning on releasing anything this time around.

That is a crying shame, because my system could do-er...  :(

**Comment by flarn2006** [comment]
Hopefully they have a better excuse than last time.

**Comment by ljlueloff** [comment]
> Hopefully they have a better excuse than last time.

@flarn2006 You mean the....ooohhhh we created something scary and have soggy diapers excuse with GPT-3?

**Comment by ljlueloff** [comment]
@flarn2006 If they don't make excuses or drag their feet, and I finish my system build in a relatively congruent time frame, hopefully I can help...

**Comment by fen0s** [comment]
A little update: OpenAI's now running their own API with GPT-3 on it. https://beta.openai.com
You can apply for access, but seems like they're aiming mostly for big companies, not researches. Sad, way too sad.

**Comment by stefan-it** [comment]
But who put the "Open" in OpenAI then 🤔

**Comment by yassineAlouini** [comment]
I guess we will need to "fundraise" enough GPU-compute to run the GPT3 model. :smile:

**Comment by fen0s** [comment]
It should be possible to run lower-models on regular GPUs, like 1b model. But we don't have the model itself, and seems that OpenAI is against releasing it and would rather commercialize it :(

**Comment by StealthySemicolon** [comment]
I wonder if you could hardcode the 175B model into an electronic chip(like an ASIC but more specific)

**Comment by yassineAlouini** [comment]
> I wonder if you could hardcode the 175B model into an electronic chip(like an ASIC but more specific)

Very interesting as an idea. @StealthySemicolon do you have reference to other similar work done in the past?

**Comment by StealthySemicolon** [comment]
> > I wonder if you could hardcode the 175B model into an electronic chip(like an ASIC but more specific)
> 
> Very interesting as an idea. @StealthySemicolon do you have reference to other similar work done in the past?

No, just a hunch. Even if I did know how to do this, it's not like OpenAI would publicly release the model weights...

**Comment by shashankMadan-designEsthetics** [comment]
Guys when is this gonna be integrated!?

**Comment by fen0s** [comment]
When OpenAI decides to release GPT-3 open-sourcely, but this won't happen it seems, they just want to sell access to big corporations.

**Comment by shashankMadan-designEsthetics** [comment]
https://bdtechtalks.com/2020/08/17/openai-gpt-3-commercial-ai/amp/

**Comment by bhack** [comment]
https://arxiv.org/abs/2009.07118
https://github.com/timoschick/pet

**Comment by OverlordQ** [comment]
> Hopefully they have a better excuse than last time.

Because Microsoft [gave us money.](https://openai.com/blog/openai-licenses-gpt-3-technology-to-microsoft/)

**Comment by Clickative** [comment]
GPT-3 is not coming out anytime soon :(

**Comment by shashankMadan-designEsthetics** [comment]
this thread signifies capitalism's pros and cons at the same time...😅

**Comment by albusdemens** [comment]
> The full model will be at least 350 GB (16-bit parameters). You'd need to partition it across more than (350 GB) / (16 GB) ~ **22 GPUs** just to run it! Not to mention the egress costs of making a model that size available.
> 
> Of course, the paper shows **8** different-sized models, **4 of which are smaller than GPT-2**, so some of those could be practical. 🙂
> 
> ![image](https://user-images.githubusercontent.com/5590173/83222642-ca2dc000-a146-11ea-934d-58001e955941.png)

Is there a way to estimate the size of the GPT-3 XL model?

**Comment by PinkDev1** [comment]
we're still waiting.. :(

**Comment by srulikbd** [comment]
it seems that a replication of GPT3 might be open source soon!! :
https://www.eleuther.ai/
https://github.com/EleutherAI

**Comment by flarn2006** [comment]
Nice! Hope that works out!

**Comment by NielsRogge** [comment]
Closing this as GPT-3 won't be open-sourced unfortunately.

Have a look at an open-source effort (a 176-billion parameter multilingual language model called BLOOM) to replicate it here:
* blog post: https://bigscience.huggingface.co/blog/bloom
* model: https://huggingface.co/bigscience/bloom.

Besides that, [EleutherAI](https://huggingface.co/EleutherAI) and other groups (such as [this one](https://github.com/THUDM/GLM-130B)) have been working on several open-source variants of GPT-3.

**Comment by Yusuf-YENICERI** [comment]
Don't worry, if they made it, some other people going to make it, inshaAllah.

There are already replications, so wait for that.

**Comment by sarahwang93** [comment]
anyone told me is there GPT-3 available?  the official one

**Comment by Yusuf-YENICERI** [comment]
@sarahwang93 No. It's not open sourced and they won't probably. Because they are able to make money using that. 

Replying to myself: Yes you are right. Other people made millions of it, Alhamdulillah.

**Comment by sarahwang93** [comment]
@Yusuf-YENICERI Hope that they could opensource after they made enough money, my phd dissertation is waiting for it.

**Comment by Yusuf-YENICERI** [comment]
@sarahwang93  Why do you need it? You can't run it simply. It's a real huge model. Maybe 700GB VRAM required to run it. If you want to know about how its made you can check the paper of it.

There are other open source models. You may want to check them.

**Comment by NielsRogge** [comment]
There's also the Open LLM leaderboard which benchmarks all openly available LLMs on 4 benchmarks: https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard. 

Of course this is not perfect as it only includes 4 benchmarks, but is still gives a nice overview of the best open-source LLMs out there.

**Comment by Yusuf-YENICERI** [comment]
@NielsRogge 
https://chat.lmsys.org/?arena
This is better for simplicity, InshaAllah.

**Comment by Ispendedfourm** [comment]
Well, guys, I think we haven't chatted since 2023, so I am ruining it here in 2025.
